lib/class-wp-rest-sidebars-controller.php

```diff
@@ -0,0 +1,342 @@
+ *
+ * @package WP_API_Sidebars\Controllers
+ */
+class WP_REST_Sidebars_Controller extends WP_REST_Controller {
+
+	/**
+	 * Plugins controller constructor.
+	 *
+	 * @since 5.5.0
+	 */
+	public function __construct() {
+		$this->namespace = '__experimental';
+		$this->rest_base = 'sidebars';
+	}
+
+	/**
+	 * Registers the controllers routes
+	 *
+	 * @return null
+	 */
+	public function register_routes() {
+		// lists all sidebars
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base, [
+			[
+				'methods'  => WP_REST_Server::READABLE,
+				'callback' => [ $this, 'get_items' ],
+			],
+		] );
+
+		// lists a single sidebar based on the given id
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[\w-]+)',
+			[
+				[
+					'methods'  => WP_REST_Server::READABLE,
+					'callback' => [ $this, 'get_item' ],
+					'args'     => [
+						'id' => [
+							'description'       => 'The id of a registered sidebar',
+							'type'              => 'string',
+							'validate_callback' => function ( $id ) {
+								return ! is_null( self::get_sidebar( $id ) );
+							},
+						],
+					],
+				],
+			]
+		);
+
+		// lists a single sidebar based on the given id
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[\w-]+)',
+			[
+				[
+					'methods'  => WP_REST_Server::EDITABLE,
+					'callback' => [ $this, 'update_item' ],
+					'args'     => [
+						'id' => [
+							'description'       => 'The id of a registered sidebar',
+							'type'              => 'string',
+							'validate_callback' => function ( $id ) {
+								return ! is_null( self::get_sidebar( $id ) );
+							},
+						],
+					],
+				],
+			]
+		);
+	}
+
+	public function update_item( $request ) {
+		global $wp_registered_widget_updates;
+		$sidebar_id    = $request->get_param( 'id' );
+		$input_widgets = $request->get_param( 'widgets' );
+
+		if ( ! is_array( $input_widgets ) ) {
+			return;
+		}
+
+		$numbers = [];
+		foreach ( $wp_registered_widget_updates as $id_base => $control ) {
+			$numbers[ $id_base ] = $control['callback'][0]->number + 1;
+		}
+
+		$created_widgets = [];
+
+		// Create and update widgets
+		$sidebar_widgets_ids = [];
+		foreach ( $input_widgets as $input_widget ) {
+			if ( ! isset( $wp_registered_widget_updates[ $input_widget['id_base'] ] ) ) {
+				continue;
+			}
+			$update_control = $wp_registered_widget_updates[ $input_widget['id_base'] ];
+			if ( ! isset( $input_widget['id'] ) ) {
+				$number = $numbers[ $input_widget['id_base'] ] ++;
+				$id     = $input_widget['id_base'] . '-' . $number;
+
+				$input_widget['id']     = $id;
+				$input_widget['number'] = $number;
+				$created_widgets[]      = [
+					'id'      => $id,
+					'number'  => $number,
+					'id_base' => $input_widget['id_base'],
+				];
+			}
+
+			$_POST = $input_widget;
+
+			$_POST[ 'widget-' . $input_widget['id_base'] ][ $input_widget['number'] ] = $input_widget['settings'];
```

Comment:
This needs to be slashed. REST API request data is unslashed, but the PHP globals are slashed.

Reply:
Good spot! It seems like [class-wp-widget.php](https://github.com/WordPress/WordPress/blob/master/wp-includes/class-wp-widget.php#L433) immediately calls `stripslashes_deep` on that data so it should go in there in the slashed form.